### ISSUE TITLE
Publishing changes from background threads is not allowed; make sure to publish values from the main thread (via operators like receive(on:)) on model updates.

### ISSUE THREAD

**Issue by jxhug** [sindresorhus/Defaults#152]
When using Defaults, I get this error in the SwiftUI file.
<img width="954" alt="image" src="https://github.com/sindresorhus/Defaults/assets/72482214/a073e635-24dd-4835-b107-bd590532123b">
it's on line 16: 
<img width="362" alt="image" src="https://github.com/sindresorhus/Defaults/assets/72482214/fa6144b0-c4fc-47bb-a9c8-8a19b2d9066c">
I am using the main branch for the dependency rule.

**Comment by sindresorhus** [comment]
I don't think this is a problem with Defaults. It is correctly marked as `@MainActor`, but that annotation only works when called from a concurrency context. So somewhere you are calling it from a non-main-actor context.
